lib/i18n_yaml_editor/transformation.rb

```diff
@@ -37,5 +37,25 @@ def nest_hash hash
       result
     end
     module_function :nest_hash
+
+    ##
+    # Replaces from_locale with to_locale in filename of path
+    # Supports filenames with locale prefix or suffix
+    #
+    # @param [String] from_locale
+    # @param [String] to_locale
+    # @param [String] path
+    #
+    # @example Get path for en locale path from existing dk locale path
+    #   Transformation.replace_locale_in_path('dk', 'en', '/tmp/dk.foo.yml') #=> "/tmp/en.foo.yml"
+    #   Transformation.replace_locale_in_path('dk', 'en', '/tmp/foo.dk.yml') #=> "/tmp/foo.en.yml"
+    #
+    # @return [String]
+    def replace_locale_in_path(from_locale, to_locale, path)
+      parts = File.basename(path).split('.')
+      parts[parts.index(from_locale)] = to_locale
+      File.join(File.dirname(path), parts.join('.'))
+    end
```

Comment:
I like to use IYE not only through the browser frontend, but in migration scripts etc. too. So having the `nest_hash` and `flatten_hash` methods available was pretty useful and the same applies to this new method in my opinion.
Besides, why not make it public? It is documented and tested.